test/blackbox/run_blackbox_tests/test_cases_autoconnect.c

```diff
@@ -0,0 +1,153 @@
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+/* Modify this to change the logging level of Meshlink */
+#define TEST_MESHLINK_LOG_LEVEL MESHLINK_DEBUG
+
+#include "execute_tests.h"
+#include "test_cases_autoconnect.h"
+#include <pthread.h>
+#include "../common/test_step.h"
+#include "../common/common_handlers.h"
+#include "../../utils.h"
+#include <stdlib.h>
+#include <stdarg.h>
+#include <setjmp.h>
+#include <cmocka.h>
+#include <assert.h>
+#include <string.h>
+
+static void test_case_autoconnect(void **state);
+static bool test_steps_mesh_autoconnect(void);
+
+/* State structure for meshlink_blacklist Test Case #1 */
+static black_box_state_t test_mesh_autoconnect_state = {
+	.test_case_name = "test_case_mesh_autoconnect",
+};
+struct sync_flag test_autoconnect_blacklisted = {.mutex  = PTHREAD_MUTEX_INITIALIZER, .cond = PTHREAD_COND_INITIALIZER, .flag = false};
+
+/* Execute meshlink_blacklist Test Case # 1*/
+void test_case_autoconnect(void **state) {
+	execute_test(test_steps_mesh_autoconnect, state);
+}
+
+void callback_logger(meshlink_handle_t *mesh, meshlink_log_level_t level,
+                     const char *text) {
+	(void)mesh;
+	(void)level;
+
+	fprintf((FILE *)mesh->priv, "%s: {%s}\n", mesh->name, text);
+
+	if((check_sync_flag(&test_autoconnect_blacklisted) == true) && (strcmp("m1n2", mesh->name) == 0)) {
+		assert(strcmp(text, "Autoconnect trying to connect to m1n1") != 0);
+	}
+
+}
+
+static void receive(meshlink_handle_t *mesh, meshlink_node_t *src, const void *data, size_t len) {
+	const char *msg = data;
+	assert(len);
+}
+
+static bool bar_reachable;
+
+static void status_cb(meshlink_handle_t *mesh, meshlink_node_t *node, bool reachable) {
+	if(!strcmp(node->name, "bar") && reachable) {
+		bar_reachable = true;
+	}
+}
+
+
+/* Test Steps for meshlink_blacklist Test Case # 1
+
+    Test Steps:
+    1. Open both the node instances
+    2. Join bar node with foo and Send & Receive data
+    3. Blacklist bar and Send & Receive data
+
+    Expected Result:
+    When default blacklist is disabled, foo node should receive data from bar
+    but when enabled foo node should not receive data
+*/
+bool test_steps_mesh_autoconnect(void) {
+	FILE *node1, *node2;
+	char *invite = NULL;
+	meshlink_node_t *node = NULL;
+
+	meshlink_destroy("m1n1");
+	meshlink_destroy("m1n2");
+
+	// Open two new meshlink instance.
+	meshlink_handle_t *mesh1 = meshlink_open("m1n1", "m1n1", "autoconnect", DEV_CLASS_BACKBONE);
+	assert(mesh1 != NULL);
+	assert((node1 = fopen("m1n1.log", "a+")));
```

Comment:
Is it necessary to write the log output to a file? If it's written to stderr, it should still be visible, right? We don't do this for any other tests...